lib/collection/src/shards/shard_holder.rs

```diff
@@ -110,11 +150,16 @@ impl ShardHashRing {
 pub type LockedShardHolder = RwLock<ShardHolder>;
 
 impl ShardHolder {
-    pub fn new(collection_path: &Path) -> CollectionResult<Self> {
-        let rings = HashMap::from([(None, ShardHashRing::single())]);
+    pub fn new(collection_path: &Path, resharding: Option<ShardId>) -> CollectionResult<Self> {
+        let default_ring = resharding.map_or_else(ShardHashRing::single, ShardHashRing::resharding);
```

Comment:
`resharding` argument is propagated from `Collection`.

If resharding is in progress and `Collection` loaded some resharding state, then this single line will restore resharding hashring and resharding shard to the `new` hashring.

It's a bit unobvious, because `clippy` forced me to pass methods instead of closures. It's this:

```rust
let default_ring = resharding.map_or_else(
    // if `resharding` is `None` - initialize empty `single` hashring
    || ShardHashRing::single(),
    // if `resharding` is `Some(shard)` - initialize `resharding` hashring with `shard` already added to `new` hashring
    |shard| ShardHashRing::resharding(shard),
);
```